### PR TITLE
27 use proper goto controller joint and cartesian

### LINKED ISSUE #27: Use proper goto controller (joint and cartesian)

### PR DESCRIPTION
**Pull request by RemiFabre**

### CODE REVIEW

**Comment by RemiFabre** [comment]
First working version, not tested enough yet. Review appreciated since I'm not sure if this is how we want it to be done.

I suggest we stabilise a simple version (like this one), test it IRL and merge it, and then we create another issue to get the fancy features like sharing a goal handle to the sdk_client user.

Tested with this sdk_client code:
```
import time

import numpy as np

from reachy2_sdk import ReachySDK


def build_pose_matrix(x: float, y: float, z: float):
    # The effector is always at the same orientation in the world frame
    return np.array(
        [
            [0, 0, -1, x],
            [0, 1, 0, y],
            [1, 0, 0, z],
            [0, 0, 0, 1],
        ]
    )


def follow_square(reachy: ReachySDK):
    # In A position, the effector is at (0.3, -0,4, -0.3) in the world frame
    # In B position, the effector is at (0.3, -0.4, 0) in the world frame
    # In C position, the effector is at (0.3, -0.1, 0.0) in the world frame
    # In D position, the effector is at (0.3, -0.1, -0.3) in the world frame

    pose = build_pose_matrix(0.3, -0.4, -0.3)
    ik = reachy.r_arm.inverse_kinematics(pose)
    reachy.r_arm.goto_joints(ik, 2.0, degrees=True)

    pose = build_pose_matrix(0.3, -0.4, 0.0)
    ik = reachy.r_arm.inverse_kinematics(pose)
    reachy.r_arm.goto_joints(ik, 2.0, degrees=True)

    pose = build_pose_matrix(0.3, -0.1, 0.0)
    ik = reachy.r_arm.inverse_kinematics(pose)
    reachy.r_arm.goto_joints(ik, 2.0, degrees=True)

    pose = build_pose_matrix(0.3, -0.1, -0.3)
    ik = reachy.r_arm.inverse_kinematics(pose)
    reachy.r_arm.goto_joints(ik, 2.0, degrees=True)


def main_test():
    print("Trying to connect on localhost Reachy...")
    time.sleep(1.0)
    reachy = ReachySDK(host="localhost")

    time.sleep(1.0)
    if reachy.grpc_status == "disconnected":
        print("Failed to connect to Reachy, exiting...")
        return

    print("Putting each joint at 0 degrees angle")
    time.sleep(0.5)
    for joint in reachy.joints.values():
        joint.goal_position = 0

    follow_square(reachy)

    print("Going back to initial position")
    time.sleep(0.5)
    for joint in reachy.joints.values():
        joint.goal_position = 0

    print("Finished testing, disconnecting from Reachy...")
    time.sleep(0.5)
    reachy.disconnect()


if __name__ == "__main__":
    main_test()

```

File: reachy_sdk_server/reachy_sdk_server/grpc_server/joint_server.py [hunk 1]
```diff
@@ -50,6 +65,15 @@ def register_all_services(self, server: grpc.Server) -> None:
         for serv in self.services:
             serv.register_to_server(server)
 
+    def spin_asyncio(self) -> None:
```
Comment (pierre-rouanet) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/joint_server.py:76]: Similar to my comment in the other PR, I'm not sure why you need to use sync there?

File: reachy_sdk_server/reachy_sdk_server/abstract_bridge_node.py [hunk 2]
```diff
@@ -19,11 +25,14 @@
...
 class AbstractBridgeNode(Node):
-    def __init__(self, reachy_config_path: str = None) -> None:
+    def __init__(
+        self, reachy_config_path: str = None, asyncio_loop: AbstractEventLoop = None
+    ) -> None:
         super().__init__(node_name="reachy_abstract_bridge_node")
 
         self.logger = self.get_logger()
 
+        self.asyncio_loop = asyncio_loop
```
Comment (pierre-rouanet) [inline reachy_sdk_server/reachy_sdk_server/abstract_bridge_node.py:35]: why this is needed?

**Review by pierre-rouanet — COMMENTED** [review]
I haven't followed closely enough what's been done here to understand. Maybe we find some time when you are ready and we check that together?

**Pull request merged**